File: SkinEngine/src/main/java/com/lulu/skin/SkinItem.java
package com.lulu.skin;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import java.util.List;

/**
 * 皮肤收集类 Item
 */
public class SkinItem {

    private View view;

    private List<SkinAttr> attrs;


    public SkinItem(View view, List<SkinAttr> attrs) {
        this.view = view;
        this.attrs = attrs;
    }

    public void apply(Context context) {
        if (view == null || attrs == null)
            return;
        for (SkinAttr attr : attrs) {
            String attrName = attr.getAttrName();
            String attrType = attr.getAttrType();
            String resName = attr.getResName();
            int resId = attr.getResId();
            if ("background".equals(attrName)) {
                if ("color".equals(attrType)) {
                    view.setBackgroundColor(SkinEngine.get().getColor(context, resName,resId));
                } else if ("drawable".equals(attrType)) {
                    view.setBackground(SkinEngine.get().getDrawable(context, resName,resId));
                }
            } else if ("textColor".equals(attrName)) {
                if (view instanceof TextView && "color".equals(attrType)) {
                    ((TextView) view).setTextColor(SkinEngine.get().getColor(context, resName,resId));
                }
            }
        }
    }

}

File: SkinEngine/src/main/java/com/lulu/skin/SkinItem.kt
package com.lulu.skin

import android.content.Context
import android.view.View
import android.widget.TextView
import com.lulu.skin.support.SupportSkinManager
import java.lang.ref.WeakReference

/**
 * 皮肤收集类 Item
 */
class SkinItem(view: View?, private val attrs: List<SkinAttr>?) {

    /**
     * 使用弱引用 View, 防止内存泄漏
     */
    private val viewRef = WeakReference<View>(view)

    fun apply(context: Context) {
        val view = viewRef.get()
        if (view == null || attrs == null) return
        for (attr in attrs) {
            val attrName = attr.attrName
            val attrType = attr.attrType
            val resName = attr.resName
            val resId = attr.resId
            SupportSkinManager.getSupportSkinAttr(attrName)?.doAction(attrType, view, context, resName, resId)
        }
    }

}